### PR TITLE
Fixes broken avatar slug link in availability page : OG image

### LINKED ISSUE #3159: og-image not working for event-links
i believe this used to work, now it throws an error: 

image

test yourself:

https://cards-dev.twitter.com/validator

https://cal.com/peer/30min

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

- Fixes the broken avatar link in availability page (OG-image for event links)

Fixes #3159 

After:
image
Tested live

**Environment**: Staging(main branch)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -368,6 +368,10 @@ const AvailabilityPage = ({ profile, eventType }: Props) => {
     ),
     [timeZone]
   );
+  const currentPath = router.asPath.split("/");
+  currentPath.pop();
+  currentPath.shift();
+  const slug = currentPath.join("/");
```
Comment (zomars) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:374]: Will this work with queries?

Comment (alishaz-polymath) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:374]: Do we use this page with queries alone? Can you give me an example?
I would probably need to skip the execution of this and the HeadSEO code if we run this page on queries alone I guess

Comment (zomars) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:374]: The date is basically set from a query. Hence my question

**Comment by alishaz-polymath** [comment]
[Here is the Loom](https://www.loom.com/share/fac45c6f6f0a4fe8983427a411d7fae5) showing a booking flow after the changes. Hopefully this takes care of your aforementioned concern @zomars ? If you meant something else, please do let me know. 🙏

**Review by emrysal — APPROVED** [review]
Confirmed with @hariombalhara this should work with Embeds, so good to go :)

**Review by zomars — CHANGES_REQUESTED** [review]
It works on the happy path only @alishaz-polymath. There can be slashes in queries as well.

Having this URL:

https://cal-ipl1sywwd-cal.vercel.app/pro/30min?slashtest=a/b

Will produce this image:

https://og-image-one-pi.vercel.app/Meet%20**Pro%20Example**%20%3Cbr%3E%2030min.png?md=1&images=https%3A%2F%2Fcal.com%2Flogo-white.svg&images=https://cal.dev/pro/30min?slashtest=a/avatar.png

Which is busted. That was what my concern referring to.


EDIT: 
I know it's an edge case but there may be a valid `redirectUrl=someUrlwithslashes` that a user can use on their website. So it shouldn't break with that IMO.

File: apps/web/components/booking/pages/AvailabilityPage.tsx (hunk 1 above)
Comment (zomars) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:374]: [See my comment](https://github.com/calcom/cal.com/pull/3255#pullrequestreview-1034536791)

**Comment by alishaz-polymath** [comment]
> It works on the happy path only @alishaz-polymath. There can be slashes in queries as well.
> 
> Having this URL:
> 
> https://cal-ipl1sywwd-cal.vercel.app/pro/30min?slashtest=a/b
> 
> Will produce this image:
> 
> https://og-image-one-pi.vercel.app/Meet%20**Pro%20Example**%20%3Cbr%3E%2030min.png?md=1&images=https%3A%2F%2Fcal.com%2Flogo-white.svg&images=https://cal.dev/pro/30min?slashtest=a/avatar.png
> 
> Which is busted. That was what my concern referring to.

Are there any working examples of it though? I mean I get it that an extra Slash would break it, but can we actually get an extra slash in here through normal means which are useful in the app itself? As in, do we process any query parameters on this page which might contain a value containing a Slash?

**Comment by zomars** [comment]
@alishaz-polymath see my updated comment

**Review by zomars — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

**Pull request merged**